elastic/kibana
# [Enterprise Search] Present consistent Connection Error experience before and after plugin has been loaded

## Summary

1. Adds a new string value `errorConnectingMessage` to `HttpLogic`, this replaces previous the binary value `errorConnecting`.  We test for the existence of `errorConnectingMessage` in places we previously would test the truthiness of `errorConnecting`
2. Removes the `errorConnectingMessage` prop being passed down through multiple components, and instead relies on the new `HttpLogic.value.errorConnectingMessage`
3. Ensures that a consistent error message is being passed to `errorConnectingMessage` whether it is being set from as a prop or from the `errorConnectingInterceptor`

## Screenshots
![Screen Shot 2021-12-27 at 2 39 26 PM](https://user-images.githubusercontent.com/2479295/147502294-494d4c16-c063-4d16-aaff-4d237c3d569d.png)

## QA
1. Start Elasticsearch, Enterprise Search, and this branch locally
2. Load the Enterprise Search product selector in Kibana (ie `http://localhost:5601/<YOUR THREE LETTER CODE>/app/enterprise_search/overview`)
3. Stop Enterprise Search
4. Refresh the page
- [x] You should see an error connecting screen with `502 Bad Gateway` in the message box
5. Start Enterprise Search again
6. Refresh the page
- [x] You should see the product selector again

7. Open App Search in Kibana
8. Stop Enterprise Search
9. Click any link internal to App Search
- [x] You should see an error connecting screen with `502 Bad Gateway` in the message box
10. Refresh the page
- [x] You should see an error connecting screen with `502 Bad Gateway` in the message box
11. Start Enterprise Search again
12. Refresh the page
- [x] The page you attempted to navigate to should load
13. Return to the product selector

14. Open Workplace Search in Kibana
15. Stop Enterprise Search
16. Click any link internal to Workplace Search
- [x] You should see an error connecting screen with `502 Bad Gateway` in the message box
17. Refresh the page
- [x] You should see an error connecting screen with `502 Bad Gateway` in the message box
16. Start Enterprise Search again
17. Refresh the page
- [x] The page you attempted to navigate to should load

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios(https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
